app/v1/models/response/retrieveForeignPropertyAnnualSubmission/RetrieveForeignPropertyAnnualSubmissionResponse.scala

```diff
@@ -16,12 +16,30 @@
 
 package v1.models.response.retrieveForeignPropertyAnnualSubmission
 
+import config.AppConfig
 import play.api.libs.json.{Json, OFormat}
+import v1.hateoas.HateoasLinksFactory
+import v1.models.hateoas.{HateoasData, Link}
+import v1.models.response.amendForeignPropertyAnnualSubmission.AmendForeignPropertyAnnualSubmissionResponse._
 import v1.models.response.retrieveForeignPropertyAnnualSubmission.foreignFhlEea.ForeignFhlEeaEntry
 import v1.models.response.retrieveForeignPropertyAnnualSubmission.foreignProperty.ForeignPropertyEntry
 
 case class RetrieveForeignPropertyAnnualSubmissionResponse(foreignFhlEea: Option[ForeignFhlEeaEntry], foreignProperty: Option[Seq[ForeignPropertyEntry]])
 
 object RetrieveForeignPropertyAnnualSubmissionResponse {
   implicit  val format: OFormat[RetrieveForeignPropertyAnnualSubmissionResponse] = Json.format[RetrieveForeignPropertyAnnualSubmissionResponse]
-}
\ No newline at end of file
+
+  implicit object RetrieveForeignPropertyAnnualSubmissionLinksFactory extends
```

Comment:
Add tests for links please, like we have here: https://github.com/hmrc/property-business-api/blob/master/test/v1/models/response/retrieveForeignPropertyPeriodSummary/RetrieveForeignPropertyPeriodSummaryResponseSpec.scala#L184-L198